hooks/pre_validations/30-mongo_storage_engine.rb

```diff
@@ -0,0 +1,47 @@
+MONGO_DIR = '/var/lib/mongodb/'
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def disk_space
+  # Check diskspace in /var/tmp
+  logger.info 'Checking available diskspace in /var/tmp for upgrade'
+  total_space = `df -k --output=avail /var/tmp`.split("\n").last.to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  #{@MONGO_DIR}`.split[0].to_i : 0
+  if total_space < mongo_size
+    fail_and_exit "There is not enough free space #{total_space}, the size of MongoDB database is #{mongo_size}, please add additional space to /var/tmp and try again, exiting."
+  else
+    logger.debug "There is #{total_space} free space on disk, which is more than the size of the MongoDB database of #{mongo_size}, continuing with upgrade"
+  end
+end
+
+if app_value(:upgrade_mongo_storage_engine)
+  # Fail if MongoDB is already running in wiredTiger.
+  if File.file?("#{MONGO_DIR}/WiredTiger.wt")
+    fail_and_exit 'MongoDB has already been switched to wiredTiger. Exiting.'
+  end
+  if app_value(:upgrade)
+    fail_and_exit 'Concurrent use of --upgrade and --upgrade-mongo-storage-engine is not supported. '\
+                  'Please run --upgrade first, then --upgrade-mongo-storage-engine.'
+  end
+
+  # Fail if we are not using the Katello/Foreman Proxy Content scenario.
+  katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
+  foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
+  fail_and_exit 'MongoDB storage engine upgrade is not currently supported for the chosen scenario.' unless katello || foreman_proxy_content
+
+  # Fail if Katello MongoDB is not localhost, does not have a valid db connection, or an invalid value in the answerfile.
+  if katello
+    pulp_host_param = param('katello', 'pulp_db_seeds')
+    mongo_db_name = param('katello', 'pulp_db_name')
+    fail_and_exit('No value set for MongoDB connection') unless pulp_host_param
+    fail_and_exit('Unknown pulp database name') unless mongo_db_name || mongo_db_name != "pulp_database"
```

Comment:
Why bother checking the value?

Reply:
We wanted to confirm there was a value in there in case the user is using a custom DB name and didn't empty it out, more for just safety checking.

Reply:
Sorry, I don't follow. Why would we care if "the user is using a custom DB name"? I'm not sure what you mean by "didn't empty it out"?

Reply:
Ah sorry there is so much crap going on in this PR I misspoke. We are checking that to make sure there is a value so we can use that that in the pre hook as the db name, if they are using a custom name, so this is just checking to make sure a value is there.

Reply:
in that case `|| mongo_db_name != "pulp_database"` wouldn't be required. `mongo_db_name != "pulp_database"` will only get run if `mongo_db_name` is `nil` or `false`, in which case `mongo_db_name != "pulp_database"` will defiently return `true`. 

`mongo_db_name || mongo_db_name != "pulp_database"` will always return `true` so it's kind of pointless. Unless i'm missing an edge case?